Testinfra is choosing Upstart for CentOS 6 as service provider

Hello,

I've been tracking down some failing tests and I've noticed that testinfra `1.6.5` is picking up the `Upstart` service provider for my test case for CentOS 6, which feels wrong and it should be choosing the `SysvService` instead.  

Here's an example:

```
    @pytest.mark.docker_images('centos:6')
    def test_httpd_service_starts_up_through_etc_initd(host, wait_for_svc):
        svc = host.run('/etc/init.d/httpd start')
        wait_for_svc(host, 'start', timeout=5)
    
        assert svc.rc == 0
>       assert host.service('httpd').is_running
E       AssertionError: assert False
E        +  where False = <service httpd>.is_running
E        +    where <service httpd> = <class 'testinfra.modules.base.UpstartService'>('httpd')
E        +      where <class 'testinfra.modules.base.UpstartService'> = <testinfra.host.Host object at 0x7fd013d85310>.service
```

--- a/testinfra/modules/service.py
+++ b/testinfra/modules/service.py
@@ -52,7 +52,9 @@ class Service(Module):
                 and "systemd" in host.file("/sbin/init").linked_to
             ):
                 return SystemdService
-            elif host.exists("initctl"):
+            elif (host.exists("initctl")
+                    and host.exists('status')
+                    and host.file('/etc/init').is_directory):
                 return UpstartService
             return SysvService
         elif host.system_info.type == "freebsd":

